ext/util/collections.c

```diff
@@ -593,7 +593,7 @@ php_driver_user_type_set(CassUserType *ut,
   case CASS_VALUE_TYPE_TEXT:
   case CASS_VALUE_TYPE_ASCII:
   case CASS_VALUE_TYPE_VARCHAR:
-    CHECK_ERROR(cass_user_type_set_string_by_name(ut, name, Z_STRVAL_P(value)));
+    CHECK_ERROR(cass_user_type_set_string_by_name_n(ut, name, strlen(name), Z_STRVAL_P(value), Z_STRLEN_P(value)));
```

Comment:
`name` is being retrieved from the zend hash map and is [guaranteed to be a null terminated string](http://www.phpinternalsbook.com/php7/internal_types/strings/zend_strings.html#php-s-history-and-classical-c-strings); therefore the `cass_user_type_set_string_by_name_n()` is unnecessary.

Reply:
Value is user defined as far as I can tell. Generally speaking if that's the case, php strings are usually null terminated though it's not strictly guaranteed to be the case.

If it's coming from outside, the question is what the PHP intention is. Are people writing php C meant to always set it to 0, for example when referencing a string somewhere else (using strdup if necessary) or is there the potential to have it set to something else. Generally speaking guides on PHP say it's just there for a memory hack.

The original intention of always having +1 byte was apparently a hack. It's a mystery if it's always meant to be 0 and to provide safety (at least against overrunning).

If it is an issue, it would take a care combination of those two circumstances to cause a problem.